lapce-proxy/src/plugin/wasi.rs

```diff
@@ -181,30 +183,37 @@ impl PluginServerHandler for Plugin {
 
 impl Plugin {
     fn initialize(&mut self) {
-        let server_rpc = self.host.server_rpc.clone();
         let workspace = self.host.workspace.clone();
         let configurations = self.configurations.as_ref().map(unflatten_map);
-        thread::spawn(move || {
-            let root_uri = workspace.map(|p| Url::from_directory_path(p).unwrap());
-            let _ = server_rpc.server_request(
-                Initialize::METHOD,
-                #[allow(deprecated)]
-                InitializeParams {
-                    process_id: Some(process::id()),
-                    root_path: None,
-                    root_uri,
-                    capabilities: ClientCapabilities::default(),
-                    trace: None,
-                    client_info: None,
-                    locale: None,
-                    initialization_options: configurations,
-                    workspace_folders: None,
-                },
+        let root_uri = workspace.map(|p| Url::from_directory_path(p).unwrap());
+        if let Ok(value) = self.host.server_rpc.server_request(
```

Comment:
I suppose this can be turned into a non-blocking request by spawning another thread and wrapping `server_capabilities` inside an `Arc<Mutex<T>>`